Library/Homebrew/utils.rb

```diff
@@ -498,11 +498,19 @@ def pretty_ratelimit_reset(reset)
 
   class AuthenticationFailedError < Error
     def initialize(error)
-      super <<-EOS.undent
-        GitHub #{error}
-        HOMEBREW_GITHUB_API_TOKEN may be invalid or expired, check:
+      message = "Github #{error}\n"
+      if ENV["HOMEBREW_GITHUB_API_TOKEN"]
+        message << <<-EOS.undent
+          HOMEBREW_GITHUB_API_TOKEN may be invalid or expired; check:
           #{Tty.em}https://github.com/settings/tokens#{Tty.reset}
-      EOS
+        EOS
+      else
+        message << <<-EOS.undent
+          You may have old Github credentials saved in the OS X Keychain
```

Comment:
Maybe just worth saying ```
Your GitHub credentials in the OS X keychain are invalid.
Clear them with printf "protocol=https\nhost=github.com\n" | git credential-osxkeychain erase
```

Alternatively, I like the idea of perhaps just retrying the API call without the token? Thoughts?